src/mmw/apps/home/views.py

```diff
@@ -42,9 +44,12 @@ def model(request, proj_id=None):
     csrf_token.update(csrf(request))
 
     if proj_id:
-        get_object_or_404(Project, user=request.user.id, id=proj_id)
+        project = get_object_or_404(Project, id=proj_id)
 
-    return render_to_response('home/home.html', csrf_token)
+    if project.user.id == request.user.id or project.is_private is False:
```

Comment:
Also note that `project` does not always exist here. If no `proj_id` is specified, as in [http://localhost:8000/model/](http://localhost:8000/model/), this line causes a crash.